uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/leave-request-popup-controller_test.js

```diff
@@ -699,6 +699,91 @@
         });
       });
 
+      describe('when user edits leave request', function () {
+        beforeEach(function () {
+          var status = optionGroupMock.specificValue('hrleaveandabsences_leave_request_status', 'value', '3');
+          var leaveRequest = LeaveRequestInstance.init(mockData.findBy('status_id', status));
+          leaveRequest.contact_id = CRM.vars.leaveAndAbsences.contactId.toString();
+          var directiveOptions = {
+            contactId: leaveRequest.contact_id, //owner's contact id
+            leaveRequest: leaveRequest
+          };
+
+          initTestController(directiveOptions);
+        });
+
+        describe('on initialization', function () {
+          var waiting_approval;
+
+          beforeEach(function () {
+            waiting_approval = optionGroupMock.specificObject('hrleaveandabsences_leave_request_status', 'value', '3');
+          });
+
+          it('should set role to owner', function () {
+            expect($ctrl.role).toEqual('owner');
+          });
+
+          it('should set all leaverequest values', function () {
+            expect($ctrl.leaveRequest.contact_id).toEqual('' + CRM.vars.leaveAndAbsences.contactId);
+            expect($ctrl.leaveRequest.type_id).toEqual(jasmine.any(String));
```

Comment:
Why don't you use the actual values you have in the `leaveRequest` object? You are obviously testing if all the values are stored correctly in the controller, so testing for "any" string doesn't serve any purpose